index.ts

```diff
@@ -0,0 +1,58 @@
+import * as extend from 'extend';
+import * as _request from 'request';  // for types only
+import * as request from 'retry-request';
+
+const BASE_URL = 'http://metadata.google.internal/computeMetadata/v1';
+
+export type Options = _request.CoreOptions&{property?: string, uri?: string};
+
+export type Callback =
+    (error: Error|null, response?: _request.RequestResponse,
+     metadataProp?: string) => void;
+
+export function _buildMetadataAccessor(type: string) {
+  return function metadataAccessor(
+      options: string|Options|Callback, callback?: Callback) {
+    if (typeof options === 'function') {
+      callback = options;
+      options = {};
+    }
+
+    if (typeof options === 'string') {
+      options = {property: options};
+    }
+
+    let property = '';
+    if (typeof options === 'object' && options.property) {
+      property = '/' + options.property;
+    }
+
+    const reqOpts = extend(
+                        true, {
+                          uri: BASE_URL + '/' + type + property,
+                          headers: {'Metadata-Flavor': 'Google'}
+                        },
+                        options) as _request.Options &
+        {property?: string};
+    delete reqOpts.property;
+
+    const retryRequestOpts = {noResponseRetries: 0};
+
+    return request(reqOpts, retryRequestOpts, (err, res, body) => {
+      if (callback) {  // for type safety; this should already always be true
```

Comment:
Warning: this is a lame nit pick, but I think we need another commit to kick the CI server awake (I just changed a setting so that it runs on PRs).

Here it is!

```diff
- if (callback) {  // for type safety
+ if (callback) { // for type safety
```

Reply:
This particular change is enforced by `clang-format`... I personally prefer single space as well, but CI will actually fail if I make this change.

I did a force push and it seems to have triggered CI.

Reply:
Ah, okay. Thanks for the explanation.